From f9d090586dd64c90271782a2cb738e1ab0b720f2 Mon Sep 17 00:00:00 2001
From: Ben Hutchings <ben@decadent.org.uk>
Date: Fri, 10 Jun 2011 04:30:04 +0100
Subject: [PATCH] x86 idle: EXPORT_SYMBOL(default_idle, pm_idle) if
 CONFIG_APM_MODULE (only)

Commit 06ae40ce073daf233607a3c54a489f2c1e44683e ('x86 idle:
EXPORT_SYMBOL(default_idle, pm_idle) only when APM demands it') made
these exports dependent on CONFIG_APM_MODULE && CONFIG_APM_CPU_IDLE.
However, the apm module always needs these symbols.
CONFIG_APM_CPU_IDLE only controls whether it installs its CPU idle
handler by default.

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
 arch/x86/kernel/process.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/x86/kernel/process.c b/arch/x86/kernel/process.c
index 2e4928d..e1ba8cb 100644
--- a/arch/x86/kernel/process.c
+++ b/arch/x86/kernel/process.c
@@ -337,7 +337,7 @@ EXPORT_SYMBOL(boot_option_idle_override);
  * Powermanagement idle function, if any..
  */
 void (*pm_idle)(void);
-#if defined(CONFIG_APM_MODULE) && defined(CONFIG_APM_CPU_IDLE)
+#ifdef CONFIG_APM_MODULE
 EXPORT_SYMBOL(pm_idle);
 #endif
 
@@ -399,7 +399,7 @@ void default_idle(void)
 		cpu_relax();
 	}
 }
-#if defined(CONFIG_APM_MODULE) && defined(CONFIG_APM_CPU_IDLE)
+#ifdef CONFIG_APM_MODULE
 EXPORT_SYMBOL(default_idle);
 #endif
 
-- 
1.7.5.3

